### ISSUE TITLE
Proposal for SIG/WG CI Robots

### ISSUE THREAD

**Issue by mkorbi** [cdfoundation/toc#77]
# SIG/WG CI Robots
## Intro
CICD pipelines integrate the technical process and can reach from the code commit to the production deployment. Most often form follows function especially in corporations where many processes need additional integration. Also, many outputs of a CICD are spread around various systems e.g. for testing, compliance/policy conformance and so on. However, there are extensions/supporting tools serving the pipelines before & after:
- [Probot](https://github.com/probot/probot)
- [K8s-CI-Robot](https://github.com/k8s-ci-robot) as part of the CI
- [Robotframework](https://robotframework.org/)
- [Lighthouse](https://github.com/jenkins-x/lighthouse)
- [BotKube](https://www.botkube.io/)

## Issues
What I see often in the last projects are teams trying to implement tools or write their own (scripts) supporting this kind of approach. 

## Purpose
###Overview
The SIG/WG can work on a Robot Interface for Git and CI tools to allow a common integration and steering capability. It mainly focuses on tasks before and after the pipelines running, however, they can also develop the capability to intercept processes and request further input or give output.

This could be also a place for stuff like ChatOps. 

### Objectives
The group creates designs, specifications, shared code, and processes to enable the human <-> CI/CD interaction. 
 
Initial steps:

- Identify thought leadership (e.g. ProwBot) and further current implementations
- Identify relevant processes and interactions (e.g. labeling, notification, retrigger, chatops)
- Clarify terminology and define a specification

**Comment by tequilarista** [comment]
I love the idea of this, and also wondering how much this could also overlap with the interoperability SIG (not necessarily with these two SIGs having competing goals but how they might complement each other)

**Comment by mkorbi** [comment]
@fdegir yes we can do so, happy to discuss it

**Comment by fdegir** [comment]
@mkorbi Great! Added the topic to the agenda.

https://hackmd.io/@cdf-sig-interoperability/ry3TTB5DL

**Comment by fdegir** [comment]
Thanks for joining the meeting and the discussion @mkorbi!

If anyone wants to take a look at the discussion, minutes of the meeting on April 30th are available in [SIG Interoperability](https://github.com/cdfoundation/sig-interoperability/blob/master/docs/meetings.md#april-30-2020) repository under the topic Discussion on proposed SIG/WG CI Robots. (recording of the meeting should be available on Youtube in future as well.)

**Comment by mkorbi** [comment]
updated the proposal to relflect the feedback
happy to hear more!

**Comment by dlorenc** [comment]
@mkorbi were you able to attend and discuss this with SIG Interoperability yet?

**Comment by mkorbi** [comment]
@dlorenc not yet, just drop them a message

**Comment by mkorbi** [comment]
introduced the topic to the SIG Interoperability and we want to proceed by going into detailed discussions with the WG Events in CI/CD

**Comment by fdegir** [comment]
Thanks for joining the meeting @mkorbi 

Sharing the link to meeting minutes as a reference: https://hackmd.io/HuufSDMaTPyb3qxkyBKg3A#June-11-2020

**Comment by dlorenc** [comment]
Thanks Fatih! It's not clear from the minutes what the outcome of that discussion was, can you summarize here?

**Comment by mkorbi** [comment]
@dlorenc we bring this topic to the SIG Interoperability and their workstream from Events in CICD. I had today a meeting with Andreas and we agree that this topic pretty much fits with the purpose of the SIG and its workstreams and potentially extends the dimension of the current target.

**Comment by fdegir** [comment]
@mkorbi we had a presentation about Lighthouse from @abayer during the meeting today and I wanted to get you in touch with him since you referred to Lighthouse in this issue.

**Comment by abayer** [comment]
👋 Hi!

**Comment by tracymiranda** [comment]
This SIG did not move forward, closing this issue but please feel free to re-open if renewed interest in finding a path forward.
